src/mainloop.rkt

```diff
@@ -396,15 +396,38 @@
   (extract!))
 
 (define (pareto-regimes sorted ctx)
-  (let loop ([alts sorted])
+  (define branch-exprs
+    (if (flag-set? 'reduce 'branch-expressions)
+        (exprs-to-branch-on sorted ctx)
+        (context-vars ctx)))
+  (define err-lsts (batch-errors (map alt-expr sorted) (*pcontext*) ctx))
+  (define init-options (for/list ([bexpr branch-exprs]) (option-on-expr sorted err-lsts bexpr ctx)))
```

Comment:
I understand why you do the first regimes pass separately, but what if we instead set init-options to, like, `#f` and `init-errs` to, like, `-1`? So that on the first iteration we're forced to recompute everything?